#! /bin/sh /usr/share/dpatch/dpatch-run
## 14-kfreebsd-compile.dpatch by Luigi Gangitano <luigi@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix build on kfreebsd/hurd

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' squid3~/compat/fdsetsize.h squid3/compat/fdsetsize.h
--- squid3~/compat/fdsetsize.h	2010-04-01 02:41:02.000000000 +0200
+++ squid3/compat/fdsetsize.h	2010-04-12 10:45:25.000000000 +0200
@@ -33,7 +33,7 @@
  * __FD_SETSIZE. Ofcourse a user program may NEVER include bits/whatever.h
  * directly, so this is a dirty hack!
  */
-#if defined(_SQUID_LINUX_)
+#if defined(_SQUID_LINUX_) || defined (__GLIBC__)
 #undef CHANGE_FD_SETSIZE
 #define CHANGE_FD_SETSIZE 0
 #include <features.h>
